src/main/java/org/mskcc/smile/service/impl/LabelGenMessageHandlingServiceImpl.java

```diff
@@ -296,21 +310,28 @@ public void run() {
                 try {
                     SampleMetadata sample = cmoSampleLabelUpdateQueue.poll(100, TimeUnit.MILLISECONDS);
                     if (sample != null) {
-                        List<SampleMetadata> existingSamples =
-                                getExistingPatientSamples(sample.getCmoPatientId());
-
-                        // generate new cmo sample label and update sample metadata object
-                        String newCmoSampleLabel =
-                                cmoLabelGeneratorService.generateCmoSampleLabel(sample, existingSamples);
-
-                        // check if matching sample found and determine if label actually needs updating
-                        // or if we can use the same label that is already persisted for this sample
-                        // note that we want to continue publishing to the IGO_SAMPLE_UPDATE_TOPIC since
-                        // there might be other metadata changes that need to be persisted that may not
-                        // necessarily affect the cmo label generated
-                        String resolvedCmoSampleLabel = resolveAndUpdateCmoSampleLabel(sample.getPrimaryId(),
-                                existingSamples, newCmoSampleLabel);
-                        sample.setCmoSampleName(resolvedCmoSampleLabel);
+                        if (sample.getStatus().getValidationStatus()) {
+                            List<SampleMetadata> existingSamples =
+                                    getExistingPatientSamples(sample.getCmoPatientId());
+
+                            // generate new cmo sample label and update sample metadata object
+                            String newCmoSampleLabel =
+                                    cmoLabelGeneratorService.generateCmoSampleLabel(sample, existingSamples);
+                            if (newCmoSampleLabel == null) {
+                                Status newSampleStatus = cmoLabelGeneratorService
+                                        .generateSampleStatus(sample, existingSamples);
+                                sample.setStatus(newSampleStatus);
```

Comment:
just to confirm, this would also be updating the `validationStatus` to `false` right?